### ISSUE TITLE
reference style links don't work

### ISSUE THREAD

**Issue by jakov** [tanakahisateru/js-markdown-extra#6]
```
Reference-style labels (titles are optional):
An [example][id]. Then, anywhere  
else in the doc, define the link:

  [id]: http://example.com/  "Title"
```

The above works fine, but when you remove the `[example]` and just leave the `[id]` it doesn't work for some strange reasons.
I've tested it in your preview website (http://tanakahisateru.github.com/js-markdown-extra/demo.html), and it doesn't work there either. I found the corresponding code, but i can't figure out why it doens't work:

```
function _DoAnchors_reference_callback( $0, $1, $2, $3 ) {
        var whole_match = $1;
        var link_text = $2;
        var link_id = $3.toLowerCase( );
        var result = "";

        if( link_id == "" ) {
            link_id = link_text.toLowerCase( );
        }
...
```

**Comment by jakov** [comment]
By the way, it seems that `[id][]` works fine.

**Comment by tanakahisateru** [comment]
Only `[id][]` is right syntax for implicite link text if according to:
http://daringfireball.net/projects/markdown/syntax#link
PHP markdown was originally so, I think.

But now, most of modern parsers are supporting shortcut style `[id]` and also PHP markdown has supported it later.
https://github.com/michelf/php-markdown/blob/master/markdown.php#L702

I should port missing corresponding codes from current implementation of PHP markdown.

**Comment by jakov** [comment]
Oh, I didn't even notice, that it was off the original specifications. (Showdown seems to parse them.) It makes some sense to have only `[reference][]` as links, so you can still use `[square brackets]` for any other purpose without having to escape them with backslashes.

However, i think it should be in the code, at least optionally, so that one can easily activate it.

**Comment by tanakahisateru** [comment]
Branch 1.2 was merged to master. Currently it seems fixed.
someone check it correct or not please.

**Comment by tanakahisateru** [comment]
Older style link was broken. Fixed on 8b727572956600ae9e4d1b8c9969a12a9ff1c4e2 .

**Comment by tanakahisateru** [comment]
- [Text][id] ... OK
- [id][] ... OK
- [id] ... OK

It seems fixed then I close this issue.
